packages/content/src/Application/PropertyResolver/Resolver/SmartContentPropertyResolver.php

```diff
@@ -28,6 +30,8 @@ class SmartContentPropertyResolver implements PropertyResolverInterface
 {
     public function __construct(
         private RequestStack $requestStack,
+        private RequestAnalyzerInterface $requestAnalyzer,
+        private ?TargetGroupStoreInterface $targetGroupStore,
```

Comment:
Lets discuss this tomorrow. I think smart content should not have knowledge about page specific things (webspace, segments, ...).

Reply:
as discussed we will create a:

```
    /**
     * @var iterable<SmartContentFilterVisitorInterface>
     */
    public function __construct(

        private RequestStack $requestStack,

        private iterable $requestAnalyzer,
```